docs/testing_mbed_OS_5.md

```diff
@@ -0,0 +1,187 @@
+```
+myproject/TESTS/test_group/test_case_1
+```
+
+In this example, `myproject` is the project root and all the source files under the `test_case_1` directory will be included in the test. Any other source files from the OS, libraries, and your project that apply to your target's configuration will also be included in the build of your test.
+
+### Test discovery
+
+Since test cases can exist throughout a project, the tools must find them in your project's file structure before building them. This is done by searching for paths that match the pattern detailed above in the [Test code structure](#test-code-structure) section.
+
+Test discovery also obeys the same rules that is used when building your project. This means that tests that are placed under a folder with a prefix like `TARGET_`, `TOOLCHAIN_`, or `FEATURE_` will only be discovered, built, and ran if your current configuration matches this prefix.
+
+For example, if a test was placed under the folder `FEATURE_BLE` with the following path:
+
+```
+myproject/mbed-os/features/FEATURE_BLE/TESTS/ble_tests/unit_test
+```
+
+This test case would only be discovered if the target that was being testing supported the BLE feature. Otherwise, the test will be ignored.
+
+Generally, a test should not be placed under a `TARGET_` or `TOOLCHAIN_` folder, since most tests should be designed to work for all target and toolchain configurations.
+
+Tests can also be completely ignored by using the `.mbedignore` file descirbed [here]()
+
+### Test names
+
+A test case is named from its position in your project's file structure. For instance, in the above example, a test case with the path `myproject/TESTS/test_group/test_case_1` would be named `tests-test_group-test_case_1`. You will notice that the name is created by joining the directories that make up the path to the test case with a "dash" (`-`) character. This will be a unique name to identify the test case. You will see this name used throughout the build and testing process.
+
+## Building tests
+
+Tests can be built easily through mbed CLI. For information on using mbed CLI, please see its documentation.
+
+When tests are built for a target and a given toolchain, the available tests are first discovered, then built in series. You can also create a "test specification" file, which can be used by our testing tools to run automated hardware tests. For more information on the test specification file, please see the docmentation [here](https://github.com/ARMmbed/greentea#test-specification-json-formatted-input).
+
+### Building process
+
+The process for building tests is handled by the `test.py` script (not to be confused with `tests.py`) located under the `tools` directory. This handles the discovery and building of all tests cases for a given target and toolchain.
```

Comment:
I agree, my understanding is the `tests.py` file should be phased out eventually